flytekit/remote/remote.py

```diff
@@ -902,13 +902,16 @@ def upload_file(
         extra_headers.update(upload_location.headers)
         encoded_md5 = b64encode(md5_bytes)
         with open(str(to_upload), "+rb") as local_file:
-            content = local_file.read()
-            content_length = len(content)
+            # Move to the end of the file object to get its length
+            local_file.seek(0, os.SEEK_END)
+            content_length = local_file.tell()
+            # Move back to the beginning of the file object to make it ready for reading
+            local_file.seek(0)
```

Comment:
Using [stat](https://en.wikipedia.org/wiki/Stat_(system_call)) is at least one order of magnitude faster than seeking. In python, the size of a file is exposed via https://docs.python.org/3/library/stat.html#stat.ST_SIZE.